### PR TITLE
Use Docker for local development

### PR DESCRIPTION
**Pull request by cmc333333**
This sets up a Docker for running the application locally. It uses a stock image for MySQL and extends a PHP image to include the modules we're expecting and to install composer. It sets up environment variables to imitate cloud.gov's and adds quick scripts to run `composer`, `drupal`, `drush`, and `mysql`.

This also describes the setup in our README, which should make on-boarding a bit easier.

### CODE REVIEW

**Comment by fureigh** [comment]
@cmc333333 I ran `bin/composer install` and got the following error:

```
ERROR: Version in "./docker-compose.yml" is unsupported. You might be seeing this error because you're using the wrong Compose file version. Either specify a supported version ("2.0", "2.1", "3.0", "3.1", "3.2") and place your service definitions under the `services` key, or omit the `version` key and place your service definitions at the root of the file to use version 1.
For more on the Compose file format versions, see https://docs.docker.com/compose/compose-file/
```

Docker claims to be up-to-date. Here's the complete version info in case it's useful:
```
Client:
 Version:	17.12.0-ce
 API version:	1.35
 Go version:	go1.9.2
 Git commit:	c97c6d6
 Built:	Wed Dec 27 20:03:51 2017
 OS/Arch:	darwin/amd64

Server:
 Engine:
  Version:	17.12.0-ce
  API version:	1.35 (minimum version 1.12)
  Go version:	go1.9.2
  Git commit:	c97c6d6
  Built:	Wed Dec 27 20:12:29 2017
  OS/Arch:	linux/amd64
  Experimental:	false
```

When I open `About Docker` through the Docker UI, it includes `Compose: 1.18.0`.

**Comment by cmc333333** [comment]
Ah, try bumping the docker-compose.yml version (first line of the file) to 2.2. I'll update the PR once I know the minimal version.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    build: .docker
+    ports:
+      - 8080:80
+    volumes:
+      - .:/var/www/:delegated
+    command: "/var/www/.docker/startup"
+    environment:
+      CF_INSTANCE_INDEX: '0'
+      VCAP_APPLICATION: >
+        {"name": "web"}
```
Comment (cmc333333) [inline docker-compose.yml]: Note this'll need to change if #172 isn't merged first.

**Comment by cmc333333** [comment]
Downgraded to docker compose version 3.0 and rebased to deal with a merge conflict.

**Comment by fureigh** [comment]
@cmc333333 Docker now runs and I can open a local Drupal instance, but none of the configuration has taken effect. I just get an install screen.

I see that there's a merge conflict in `bootstrap.sh`, but I doubt that's the culprit, considering that the app name in my local version of `bootstrap.sh` matches the app name in my local version of `manifest.yml`. Have you had to do anything else to get the app's configuration to load locally?

screen shot 2018-03-20 at 3 48 02 pm

**Comment by cmc333333** [comment]
Whoot whoot, thanks @fureigh !

**Pull request merged**